modules/_labs/button/react/lib/IconButton.tsx

```diff
@@ -0,0 +1,224 @@
+import {ButtonContainer} from './parts';
+
+export interface IconButtonProps extends React.ButtonHTMLAttributes<HTMLButtonElement> {
+  /**
+   * The type of the IconButton.
+   * @default IconButtonVariant.Circle
+   */
+  variant: IconButtonVariant;
+  /**
+   * The accessibility label to indicate the action triggered by clicking the IconButton.
+   */
+  'aria-label': string;
+  /**
+   * The size of the IconButton.
+   * @default ButtonSize.Medium
+   */
+  size: ButtonSize.Small | ButtonSize.Medium;
+  /**
+   * If true, toggle the IconButton on.
+   * @default false
+   */
+  toggled?: boolean;
+  /**
+   * The ref to the button that the styled component renders.
+   */
+  buttonRef?: React.Ref<HTMLButtonElement>;
+  /**
+   * The icon of the IconButton.
+   */
+  icon?: CanvasSystemIcon;
+  /**
+   * The function called when the IconButton toggled state changes.
+   */
+  onToggleChange?: (toggled: boolean | undefined) => void;
+}
+
+const IconButton = (props: IconButtonProps) => {
+  const {buttonRef, size, variant, onToggleChange, icon, toggled, children, ...elemProps} = props;
+
+  const isInitialMount = React.useRef(true);
+
+  // Only call onToggleChange on update - not on first mount
+  React.useEffect(() => {
+    if (isInitialMount.current) {
+      isInitialMount.current = false;
+    } else {
+      if (toggled && typeof onToggleChange === 'function') {
+        onToggleChange(toggled);
+      }
+    }
+  }, [toggled, onToggleChange]);
+
+  const containerStyles = {
+    padding: 0,
+    margin: variant === IconButtonVariant.Plain ? '-8px' : undefined,
+    minWidth: size === ButtonSize.Small ? spacing.l : spacing.xl, // min-width is set so buttons don't collapse in IE11
+    width: size === ButtonSize.Small ? spacing.l : spacing.xl,
+    height: size === ButtonSize.Small ? spacing.l : spacing.xl,
+    borderRadius:
+      variant === IconButtonVariant.Square || variant === IconButtonVariant.SquareFilled
+        ? borderRadius.m
+        : borderRadius.circle,
+    ['& .wd-icon']: {
+      display: 'inline-block',
+      verticalAlign: 'middle',
+      width: size === ButtonSize.Small ? '20px' : undefined,
+      height: size === ButtonSize.Small ? '20px' : undefined,
+    },
+  };
+
+  return (
+    <ButtonContainer
+      colors={getIconButtonColors(variant, toggled)}
+      size={size}
+      ref={buttonRef}
+      fillIcon={toggled}
+      extraStyles={containerStyles}
+      aria-pressed={toggled}
+      {...elemProps}
+    >
+      {icon ? <SystemIcon icon={icon} /> : children}
+    </ButtonContainer>
+  );
+};
+
+IconButton.Variant = IconButtonVariant;
+IconButton.Size = {
+  Small: ButtonSize.Small,
+  Medium: ButtonSize.Medium,
+};
+
+IconButton.defaultProps = {
+  variant: IconButtonVariant.Circle,
+  size: ButtonSize.Medium,
+};
+
+export default IconButton;
+
+const getIconButtonColors = (variant: IconButtonVariant, toggled?: boolean): ButtonColors => {
```

Comment:
ah this is so much cleaner, and passing the toggled in helps a bunch rather than returning an empty object if it's toggled on 🎉